pkg/crossnamespace/validation.go

```diff
@@ -0,0 +1,113 @@
+package crossnamespace
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	authv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/eventing/pkg/apis/feature"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/logging"
+)
+
+// type ResourceInfo struct {
+// 	Kind            string
+// 	Name            string
+// 	Namespace       string
+// 	Target          string
+// 	TargetName      string
+// 	TargetNamespace string
+// 	TargetGroup     string
+// 	FeatureStore    *feature.Store
+// }
+
+type ResourceInfo interface {
+	GetNamespace() string
+	GetName() string
+	GetTargetInfo() (kind string, name string, namespace string, group string) // target kind and target name
+}
+
+func CheckNamespace(ctx context.Context, r Resource, flag *feature.Store) *apis.FieldError {
```

Comment:
I think here we could accept a `r duckv1.KRShaped` instead of creating our own `Resource` interface since every Knative resource implements that interface already

Reply:
Actually, I think this comment might not be fully applicable as the field name is different per resource but we can reuse that interface 

```
type ResourceInfo interface {
        duckv1.KRShaped
	GetCrossNamespace() string // just an example method 
}